client/my-sites/domains/domain-management/list/domain-only-upsell-carousel.tsx

```diff
@@ -0,0 +1,126 @@
+			button_url: buttonURL,
+			source_card_type: sourceCardType,
+		} );
+	};
+
+	const renderAddSiteCard = () => {
+		const actionUrl = createSiteFromDomainOnly( domain.domain, domain.blogId );
+		return (
+			<Card className="domain-only-upsell-carousel__card" key="domain-only-upsell-site">
+				<div className="domain-only-upsell-carousel__card-wrapper is-compact">
+					<div className="domain-only-upsell-carousel__card-illustration">{ illustration }</div>
+					<div className="domain-only-upsell-carousel__card-content">
+						<div className="domain-only-upsell-carousel__card-text">
+							<h2>
+								{ translate( 'Create a site for %(domain)s', {
+									args: { domain: domain.domain },
+								} ) }
+							</h2>
+							<h3> { translate( 'Choose a theme, customize and launch your site.' ) } </h3>
+						</div>
+						<div className="domain-only-upsell-carousel__card-actions">
+							<Button
+								primary
+								href={ actionUrl }
+								onClick={ getActionClickHandler( 'primary', actionUrl, 'create-site' ) }
+							>
+								{ translate( 'Create site' ) }
+							</Button>
+						</div>
+					</div>
+				</div>
+				<TrackComponentView
+					eventName="calypso_domain_only_upsell_carousel_impression"
+					eventProperties={ { content_type: 'create-site' } }
+				/>
+			</Card>
+		);
+	};
+
+	const renderEmailCard = () => {
+		const actionUrl = emailManagementPurchaseNewEmailAccount( domain.domain, domain.domain );
+		return (
+			<Card className="domain-only-upsell-carousel__card" key="domain-only-upsell-email">
+				<div className="domain-only-upsell-carousel__card-wrapper is-compact">
+					<div className="domain-only-upsell-carousel__card-illustration">{ illustration }</div>
+					<div className="domain-only-upsell-carousel__card-content">
+						<div className="domain-only-upsell-carousel__card-text">
+							<h2>
+								{ translate( 'Add email for %(domain)s', {
+									args: { domain: domain.domain },
+								} ) }
+							</h2>
+							<h3>
+								{ translate( 'Send and receive emails from %(email)s', {
+									args: { email: `youremail@${ domain.domain }` },
+								} ) }
+							</h3>
+						</div>
+						<div className="domain-only-upsell-carousel__card-actions">
+							<Button
+								primary
+								href={ actionUrl }
+								onClick={ getActionClickHandler( 'primary', actionUrl, 'add-professional-email' ) }
+							>
+								{ translate( 'Add professional email' ) }
+							</Button>
+						</div>
+					</div>
+				</div>
+				<TrackComponentView
+					eventName="calypso_domain_only_upsell_carousel_impression"
+					eventProperties={ { content_type: 'add-professional-email' } }
+				/>
+			</Card>
+		);
+	};
+
+	const cards = [ renderAddSiteCard() ];
+	if ( ! hasPaidEmailWithUs( domain ) ) {
+		cards.push( renderEmailCard() );
+	}
+
+	return (
+		<>
+			<DotPager
+				className="domain-only-upsell-carousel"
+				hasDynamicHeight
+				showControlLabels={ false }
+				onPageSelected={ () => null }
```

Comment:
Do we need to pass a value here? I believe you can simply omit `onPageSelected` in this case.